### ISSUE TITLE
Add Calico network policy support

### ISSUE THREAD

**Issue by aaronlevy** [kubernetes-incubator/bootkube#513]
This should be "policy-only" support (leaving flannel as the pod-network provider).

See: Calico policy-only documentation here: 

http://docs.projectcalico.org/v2.2/getting-started/kubernetes/installation/hosted/kubernetes-datastore/

@abhinavdahiya is going to pick this up. I'll assign once he's part of the org.

/cc @joshrosso @yifan-gu

**Comment by rmenn** [comment]
its really funny since i worked with @abhinavdahiya, i have a working config for canal, there is something which works with canal but a `ExecStartPre=chmod 777 /etc/cni/net.d/`  which enables canal to write a file. 
 This line causes a problem and the chmod does not take effect 
https://github.com/projectcalico/cni-plugin/blob/master/k8s-install/scripts/install-cni.sh#L100

If needed i can get the logs from the server and post here, also this was possible on a previous version as i did have canal running on the server
Here is a snippet 
```
kube-system   canal-prbgf                               3/3       Running   0          22m       192.168.64.178   192.168.64.178
kube-system   canal-vkq8f                               3/3       Running   0          22m       192.168.64.179   192.168.64.179
```

**Comment by yifan-gu** [comment]
I believe this is now implemented by #524
